### PR TITLE
pkg/profilestore: Add a Parca scraper-only mode that forwards via gRPC

### LINKED ISSUE #629: Add Parca storage-less scraper
Next to having the Parca agent creating pprof profiles from eBPF and then sending it off via gRPC we should also support a mode of Parca which can scrape profiles from pprof endpoints and then immediately send them off via gRPC not having a local storage.
It's probably similar to the Prometheus agent mode.

### PR DESCRIPTION
**Pull request by metalmatze**
If there is a central place (cluster) to store profiles we still might want to collect pprof profiles and then forward them via gRPC as written in #629. 
To be able to do that we now support a `--mode=scraper` (instead of the current only `--mode=all`). 
Possibly making this a sub command instead of just a flag might be better. 

The UI is broken in --mode=scraper as only the API for the Targets page is available. We should hide the "Profiles" page in that case. Most likely we'll need to communicate the mode from the backend to the frontend for that to happen.

Closes #629

### CODE REVIEW

**Comment by metalmatze** [comment]
cc @yeya24

**Review by brancz — APPROVED** [review]
Love how simple this is thanks to everything being gRPC.

lgtm 🎉

**Comment by brancz** [comment]
Maybe the mode should be `scraper-only`? Just to make it crystal clear what the distinction is.

**Comment by metalmatze** [comment]
Yes, right? :joy: 
During the implementation (on Twitch stream) I was surprised as well that we're already done. :smile:

File: pkg/parca/parca.go [hunk 1]
```diff
@@ -108,6 +120,10 @@ func Run(ctx context.Context, logger log.Logger, reg *prometheus.Registry, flags
 		return err
 	}
 
+	if flags.Mode == "scraper" {
```
Comment (yeya24) [inline pkg/parca/parca.go:123]: Small nit, maybe make this a const?

**Review by yeya24 — COMMENTED** [review]
Amazing! I am surprised by how easy it is to implement this feature 👍 
One thing to clarify about the deployment model:
I have one central cluster and multiple edge clusters. At edge cluster, I just have to deploy scraper mode parca and sent data to a central parca, right?

File: pkg/parca/parca.go [hunk 2]
```diff
@@ -75,6 +81,12 @@ type Flags struct {
 
 	UpstreamDebuginfodServer     string        `default:"https://debuginfod.systemtap.org" help:"Upstream private/public server for debuginfod files. Defaults to https://debuginfod.systemtap.org."`
 	DebugInfodHTTPRequestTimeout time.Duration `default:"5m" help:"Timeout duration for HTTP request to upstream debuginfod server. Defaults to 5m"`
+
+	StoreAddress       string `kong:"help='gRPC address to send profiles and symbols to.'"`
+	BearerToken        string `kong:"help='Bearer token to authenticate with store.'"`
+	BearerTokenFile    string `kong:"help='File to read bearer token from to authenticate with store.'"`
+	Insecure           bool   `kong:"help='Send gRPC requests via plaintext instead of TLS.'"`
+	InsecureSkipVerify bool   `kong:"help='Skip TLS certificate verification.'"`
```
Comment (yeya24) [inline pkg/parca/parca.go:89]: Be able to specify external labels would be better. That way I can filter series by external labels.

Comment (brancz) [reply pkg/parca/parca.go:89]: 💯

File: pkg/parca/parca.go (hunk 1 above)
Comment (metalmatze) [reply pkg/parca/parca.go:123]: Yes, I was thinking the same while implementing. Sadly we still have an occurrence in the flag kong tag that won't be possible to be a constant. I'll do that anyway. :+1:

File: pkg/parca/parca.go (hunk 2 above)
Comment (metalmatze) [reply pkg/parca/parca.go:89]: Yes, totally! 
That will be a follow-up PR.

**Review by kakkoyun — APPROVED** [review]
LGTM :rocket:

**Pull request merged**